modules/integration/tests-integration/tests/src/test/java/org/wso2/carbon/integration/test/capp/CappTestCase.java

```diff
@@ -0,0 +1,122 @@
+
+        Assert.assertEquals(eventStreamManagerAdminServiceClient.getEventStreamCount(), startESCount + 2);
+        Assert.assertEquals(eventReceiverAdminServiceClient.getActiveEventReceiverCount(), startERCount + 1);
+        Assert.assertEquals(eventPublisherAdminServiceClient.getActiveEventPublisherCount(), startEPCount + 1);
+        Assert.assertEquals(eventProcessorAdminServiceClient.getAllActiveExecutionPlanConfigurationCount(), startEPCCount + 1);
+
+        try {
+            FileManager.deleteFile(cAppDirectoryPath + cAppFileName);
+        } catch (Exception e) {
+            throw new RemoteException("Exception caught when deleting the car file from CEP server", e);
+        }
+        log.info("undeploying cApp...");
+        Thread.sleep(20000);
+
+        Assert.assertEquals(eventStreamManagerAdminServiceClient.getEventStreamCount(), startESCount );
+        Assert.assertEquals(eventReceiverAdminServiceClient.getActiveEventReceiverCount(), startERCount);
+        Assert.assertEquals(eventPublisherAdminServiceClient.getActiveEventPublisherCount(), startEPCount);
+        Assert.assertEquals(eventProcessorAdminServiceClient.getAllActiveExecutionPlanConfigurationCount(), startEPCCount);
+
+    }
+
+    @Test(groups = {"wso2.cep"} ,description = "Test car file failure deployment")
+    public void testFailureCAppDeployment() throws Exception {
+
+        int startESCount  = eventStreamManagerAdminServiceClient.getEventStreamCount();
+        int startERCount  = eventReceiverAdminServiceClient.getActiveEventReceiverCount();
+        int startEPCount  = eventPublisherAdminServiceClient.getActiveEventPublisherCount();
+        int startEPCCount = eventProcessorAdminServiceClient.getAllActiveExecutionPlanConfigurationCount();
+
+        String carFilePath = FrameworkPathUtil.getSystemResourceLocation() +
+                "artifacts" + File.separator + "CEP" + File.separator + "car"
+                + File.separator;
+
+        String cAppDirectoryPath = FrameworkPathUtil.getCarbonHome() + File.separator
+                + "repository" + File.separator + "deployment" + File.separator + "server"
+                + File.separator + "carbonapps" + File.separator;
+        try {
+            FileManager.copyResourceToFileSystem(carFilePath + cAppFailureFileName, cAppDirectoryPath, cAppFailureFileName);
+        }  catch (Exception e) {
+            throw new RemoteException("Exception caught when deploying the car file into CEP server", e);
+        }
+
+        Thread.sleep(20000);
+
+        Assert.assertEquals(eventStreamManagerAdminServiceClient.getEventStreamCount(), startESCount);
+        Assert.assertEquals(eventReceiverAdminServiceClient.getActiveEventReceiverCount(), startERCount);
+        Assert.assertEquals(eventPublisherAdminServiceClient.getActiveEventPublisherCount(), startEPCount);
+        Assert.assertEquals(eventProcessorAdminServiceClient.getAllActiveExecutionPlanConfigurationCount(), startEPCCount);
+    }
```

Comment:
also remove the capp, after every test case CEP should come back to its initial state then only we can guarantee the next test will not fail!